bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/dto/Characteristic.java

```diff
@@ -0,0 +1,962 @@
+
+            default:
+                return null;
+        }
+
+        if (CoreItemFactory.NUMBER.equals(itemType) && numberSuffix != null) {
+            itemType = itemType + ":" + numberSuffix;
+            isNumberWithSuffix = true;
+        }
+
+        /*
+         * ================ CREATE FAKE PROPERTY CHANNEL =================
+         *
+         * create and return fake property channel for characteristics that
+         * are not mapped to a real channel
+         *
+         */
+        if (FAKE_PROPERTY_CHANNEL.equals(itemType)) {
+            if (value != null && value.isJsonPrimitive()) {
+                // create fake property channels for characteristics that contain only static information
+                return new ChannelDefinitionBuilder(characteristicType.toCamelCase(), FAKE_PROPERTY_CHANNEL_TYPE_UID)
+                        .withLabel(value.getAsString()).build();
+            }
+            return null;
+        }
+
+        /*
+         * ================ CREATE AND PERSIST THE CHANNEL TYPE =================
+         *
+         * NOTE: different accessories may have the same characteristicType, but
+         * their other properties e.g. min, max, step, unit may be different, so
+         * we create and persist a unique channel type ID for each characteristic
+         * instance
+         */
+        String charactersticIdentifier = characteristicType.getOpenhabType();
+        String channelTypeIdentifier = thingUID.getBridgeIds().isEmpty()
+                ? CHANNEL_TYPE_ID_FMT.formatted(charactersticIdentifier, iid, thingUID.getId(), "1")
+                : CHANNEL_TYPE_ID_FMT.formatted(charactersticIdentifier, iid, thingUID.getBridgeIds().getFirst(),
+                        thingUID.getId());
+        ChannelTypeUID channelTypeUid = new ChannelTypeUID(BINDING_ID, channelTypeIdentifier);
+        String channelTypeLabel = characteristicType.toString();
+
+        if (!isStateChannel) {
+            typeProvider.putChannelType(ChannelTypeBuilder.trigger(channelTypeUid, channelTypeLabel).build());
+
+        } else {
+            if (itemType == null) {
+                return null;
+            }
+
+            // build state description fragment
+            StateDescriptionFragmentBuilder fragBldr = StateDescriptionFragmentBuilder.create()
+                    .withReadOnly(isReadOnly);
+            if (isNumber) {
+                Optional.ofNullable(minValue).map(v -> BigDecimal.valueOf(v)).ifPresent(b -> fragBldr.withMinimum(b));
+                Optional.ofNullable(maxValue).map(v -> BigDecimal.valueOf(v)).ifPresent(b -> fragBldr.withMaximum(b));
+                Optional.ofNullable(minStep).map(v -> BigDecimal.valueOf(v)).ifPresent(b -> fragBldr.withStep(b));
+
+                if (isPercentage) {
+                    fragBldr.withPattern("%.0f %%");
+                } else if (uom != null) {
+                    fragBldr.withPattern("%.1f " + uom);
```

Comment:
If `isPercentage` is true, shouldn't `uom` then be "%" and numberSuffix "Dimensionless"? I don't fully understand how these are decoupled and the last mentioned only being set for `FILTER_LIFE_LEVEL`.